### PR TITLE
Add support for compiling templates.

### LINKED ISSUE #5: Allow custom template context
Would be great if you could allow users to specify custom context for Nunjucks through something like gulp-data.

### PR DESCRIPTION
**Pull request by d10**
Fixes #5.

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -40,3 +41,138 @@ test('support supplying custom name in a callback', function (t) {
...
+
+	stream.write(new gutil.File({
+		contents: new Buffer('foo')
+	}));
+
+	stream.end();
+});
+
+test.cb('support custom environment', function (t) {
+	var nunjucksModule = require('nunjucks');
```
Comment (d10) [inline test.js]: This seems weird; not too sure how to put it.

File: index.js [hunk 2]
```diff
@@ -31,4 +31,38 @@ module.exports = function (opts) {
 
 		cb();
 	});
-};
+}
+
+function compile(data, opts) {
+	var defaultEnv = new nunjucks.Environment();
```
Comment (sindresorhus) [inline index.js]:
```js
var env = (opts && opts.env) || new nunjucks.Environment();
```

And drop the below `env` line.

File: package.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "gulp-nunjucks",
   "version": "2.0.0",
-  "description": "Precompile Nunjucks templates",
+  "description": "Precompile/render Nunjucks templates",
```
Comment (sindresorhus) [inline package.json]: Currently you use both `render` and `compile` interchangeably to describe it for the API and docs. Can you pick one?

File: readme.md [hunk 4]
```diff
@@ -14,23 +14,57 @@ $ npm install --save-dev gulp-nunjucks
...
+		.pipe(nunjucks.compile())
+		.pipe(gulp.dest('dist'));
+});
+```
+
 
 ## API
 
-### nunjucks(options)
+### nunjucks(options) / nunjucks.precompile(options)
```
Comment (sindresorhus) [inline readme.md]:
```
### nunjucks.precompile([options])
```

File: test.js [hunk 5]
```diff
@@ -40,3 +41,138 @@ test('support supplying custom name in a callback', function (t) {
 		contents: new Buffer('<h1>{{ test }}</h1>')
 	}));
 });
+
+test.cb('have correct alias for precompile', function (t) {
+	t.is(nunjucks, nunjucks.precompile);
+	t.end();
+});
```
Comment (sindresorhus) [inline test.js]: This test isn't very useful. Remove.

File: test.js (hunk 1 above)
Comment (sindresorhus) [inline test.js]: I'm fine with that.

**Comment by d10** [comment]
@sindresorhus: updated.

**Comment by sindresorhus** [comment]
Looks good. Thank you! Happy holidays 🎄

**Comment by racse1** [comment]
Thank you guys!

**Pull request merged**